File: packages/dotnet/Db/Models/ProductSale.cs
namespace LuckyDay.Db.Models;

public class ProductSale
{
  public string Id { get; set; } = Guid.NewGuid().ToString();
  public string ProductId { get; set; } = null!;
  public string CustomerId { get; set; } = null!;
  public string UserId { get; set; } = null!;
  public int Quantity { get; set; }
  public decimal Price { get; set; }
  public DateTime DateCreated { get; set; } = DateTime.UtcNow;
  public DateTime? DateUpdated { get; set; }

  // Navigation properties
  public Product Product { get; set; } = null!;
  public Customer Customer { get; set; } = null!;
  public User User { get; set; } = null!;
}

File: apps/web/src/lib/models/product-sale.ts
export type ProductSaleModel = {
  id: string;
  productId: string;
  customerId: string;
  userId: string;
  quantity: number;
  price: number;
  dateCreated: string;
  dateUpdated: string | null;
};

export type ProductSaleCreateModel = Pick<
  ProductSaleModel,
  'productId' | 'customerId' | 'userId' | 'quantity' | 'price'
>;

export type ProductSaleUpdateModel = Pick<
  ProductSaleModel,
  'productId' | 'customerId' | 'userId' | 'quantity' | 'price'
>;

export type ProductSaleResponseModel = Pick<
  ProductSaleModel,
  | 'id'
  | 'productId'
  | 'customerId'
  | 'userId'
  | 'quantity'
  | 'price'
  | 'dateCreated'
  | 'dateUpdated'
>;
